examples/remarketing/upload_conversion_enhancement.rb

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env ruby
+#
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Adjusts an existing conversion by supplying user identifiers so Google can
+# enhance the conversion value.
+
+require 'optparse'
+require 'google/ads/google_ads'
+require 'digest'
+
+# [START upload_conversion_enhancement]
+def upload_conversion_enhancement(
+  customer_id,
+  conversion_action_id,
+  order_id,
+  conversion_date_time,
+  user_agent,
+  restatement_value,
+  currency_code
+)
+  # GoogleAdsClient will read a config file from
+  # ENV['HOME']/google_ads_config.rb when called without parameters
+  client = Google::Ads::GoogleAds::GoogleAdsClient.new
+
+  # [START create_adjustment]
+  enhancement = client.resource.conversion_adjustment do |ca|
+    ca.conversion_action = client.path.conversion_action(customer_id, conversion_action_id)
+    ca.adjustment_type = :ENHANCEMENT
+    ca.order_id = order_id
+
+    unless conversion_date_time.nil?
+      ca.gclid_date_time_pair = client.resource.gclid_date_time_pair do |pair|
+        pair.conversion_date_time = conversion_date_time
+      end
+    end
+
```

Comment:
Please add the comments around each step from [the Java example](https://github.com/googleads/google-ads-java/blob/main/google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionEnhancement.java). We added those because this process can be confusing, and calling out special cases like email address normalization is really helpful.